File: Sample/Koudai/Server/release/PyScript/Action/action1025.py
import clr, sys
clr.AddReference('ZyGames.Framework.Common');
clr.AddReference('ZyGames.Framework');
clr.AddReference('ZyGames.Framework.Game');
clr.AddReference('ZyGames.Tianjiexing.Model');
clr.AddReference('ZyGames.Tianjiexing.BLL');
clr.AddReference('ZyGames.Tianjiexing.Lang');
clr.AddReference('ZyGames.Tianjiexing.Component');

from action import *
from System import *
from System.Collections.Generic import *
from lang import Lang
from ZyGames.Framework.Common.Log import *
from ZyGames.Tianjiexing.Model.ConfigModel import *
from ZyGames.Framework.Common import *
from ZyGames.Tianjiexing.Model import *
from ZyGames.Tianjiexing.BLL import *
from ZyGames.Tianjiexing.Lang import *
from ZyGames.Framework.Game.Cache import *
from ZyGames.Framework.Game.Service import *
from ZyGames.Framework.Game.Runtime import *
from ZyGames.Framework.Common import *
from ZyGames.Framework.Cache.Generic import *
from ZyGames.Tianjiexing.Model.Config import *
from ZyGames.Tianjiexing.Model.Enum import *
from ZyGames.Tianjiexing.BLL.Base import *
from ZyGames.Tianjiexing.Component import *
from ZyGames.Framework.Common.Serialization import *


# 玩家等级升级奖励提示
class UrlParam(HttpParam):
    def __init__(self):
        HttpParam.__init__(self)

class ActionResult(DataResult):
    def __init__(self):
        DataResult.__init__(self)
        self.GoldNum = 0

def getUrlElement(httpGet, parent):
    urlParam = UrlParam();
    return urlParam;

def takeAction(urlParam, parent):
    actionResult = ActionResult();
    userId = parent.Current.User.PersonalId;
    contextUser = parent.Current.User;
    escalateInfo =  ConfigCacheSet[GeneralEscalateInfo]().FindKey(contextUser.UserLv, GeneralType.YongHu);
    if(escalateInfo):
        for info in escalateInfo.Award:
            actionResult.GoldNum = MathUtils.Addition(actionResult.GoldNum,info.Num);
        contextUser.IsLv = False;
    contextUser.GiftGold = MathUtils.Addition(contextUser.GiftGold,actionResult.GoldNum);
    return actionResult;

def buildPacket(writer, urlParam, actionResult):
    #writer.PushIntoStack(urlParam.gainNum);
    writer.PushIntoStack(actionResult.GoldNum);
    return True;

File: Sample/Koudai/Server/src/ZyGames.Tianjiexing.BLL/Action/Action1025.cs
/****************************************************************************
Copyright (c) 2013-2015 scutgame.com

http://www.scutgame.com

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
****************************************************************************/
using System;
using System.Data;
using ZyGames.Framework.Common;
using ZyGames.Tianjiexing.Model;
using ZyGames.Framework.Game.Service;
using ZyGames.Framework.Game.Cache;
using ZyGames.Tianjiexing.Component;
using ZyGames.Tianjiexing.BLL.Base;

namespace ZyGames.Tianjiexing.BLL.Action
{

    /// <summary>
    /// 1025_玩家升级提示接口
    /// </summary>
    public class Action1025 : BaseAction
    {
        private int leadership;
        private int currgeneralNum;
        private int generalNum;
        private int goldNum;


        public Action1025(ZyGames.Framework.Game.Contract.HttpGet httpGet)
            : base(ActionIDDefine.Cst_Action1025, httpGet)
        {

        }

        public override void BuildPacket()
        {
            this.PushIntoStack((short)ContextUser.UserLv);
            this.PushIntoStack(leadership);
            this.PushIntoStack(currgeneralNum);
            this.PushIntoStack(generalNum);
            this.PushIntoStack(goldNum);

        }

        public override bool GetUrlElement()
        {
            if (true)
            {
                return true;
            }
            
        }

        public override bool TakeAction()
        {
            GeneralEscalateInfo escalateInfo = new ConfigCacheSet<GeneralEscalateInfo>().FindKey(ContextUser.UserLv, GeneralType.YongHu);
            if (escalateInfo != null)
            {
                leadership = escalateInfo.Leadership;
            }
            currgeneralNum = EmbattleHelper.CurrEmbattle(ContextUser.UserID, true).Count;
            generalNum = EmbattleHelper.CurrEmbattle(ContextUser.UserID, false).Count;
            goldNum = GameConfigSet.GoldReward;
            
            return true;
        }
    }
}